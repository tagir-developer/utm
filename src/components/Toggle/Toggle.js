import React from 'react'
import classes from './Toggle.module.css'

const Toggle = ({themeToggle, darkTheme}) => {

	return (
		<div className={classes.main}>
			<input 
				type="checkbox" 
				id={classes.cbx} 
				style={{ display: 'none' }} 
				onChange={themeToggle}
				checked={darkTheme}
				special="test"
			/>
			<label htmlFor={classes.cbx} className={classes.toggle}><span></span></label>
		</div>
	)
}

Toggle.defaultProps = {
	themeToggle: () => {},
	darkTheme: false,
}

export default Toggle