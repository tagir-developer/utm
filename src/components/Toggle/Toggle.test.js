import Toggle from './Toggle'
import React from 'react'

describe('<Toggle/>', () => {

	it('Should toogle component render', () => {
		const wrapper = shallow(<Toggle />)
		expect(wrapper.find('[type="checkbox"]')).toHaveLength(1)
	})

	it('Should render checked="true" if prop "darkTheme: true"', () => {
		const wrapper = shallow(<Toggle darkTheme={true} />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})

	it('Should render checked="false" if prop "darkTheme: false"', () => {
		const wrapper = shallow(<Toggle darkTheme={false} />)
		expect(toJson(wrapper)).toMatchSnapshot()
	})

})

describe('Default props', () => {

	it('Should use default themeToggle', () => {
		const result = Toggle.defaultProps.themeToggle()
		expect(result).toBe(undefined)
	})

})