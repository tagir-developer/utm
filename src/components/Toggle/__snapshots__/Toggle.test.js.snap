// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle/> Should render checked="false" if prop "darkTheme: false" 1`] = `
<div
  className="main"
>
  <input
    checked={false}
    id="cbx"
    onChange={[Function]}
    special="test"
    style={
      Object {
        "display": "none",
      }
    }
    type="checkbox"
  />
  <label
    className="toggle"
    htmlFor="cbx"
  >
    <span />
  </label>
</div>
`;

exports[`<Toggle/> Should render checked="true" if prop "darkTheme: true" 1`] = `
<div
  className="main"
>
  <input
    checked={true}
    id="cbx"
    onChange={[Function]}
    special="test"
    style={
      Object {
        "display": "none",
      }
    }
    type="checkbox"
  />
  <label
    className="toggle"
    htmlFor="cbx"
  >
    <span />
  </label>
</div>
`;
